File: Swift/Swift3基础/11-OC&Swift互调/1-OCCallSwift/OCCallSwift/ViewController.m
//
//  ViewController.m
//  OCCallSwift
//
//  Created by 黄继平 on 2016/11/20.
//  Copyright © 2016年 PJStudio. All rights reserved.
//

#import "ViewController.h"
#import "OCCallSwift-Swift.h"


/*
    注意事项:
    1.项目名称不可以乱起;
    2.Swift中所有类/属性/方法都必须使用public或者open修饰;
    3.OC中导入"项目名-Swift.h"
 */
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    
    
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event
{
    
    CustomButton *button = [[CustomButton alloc] init];
    button.customTitle = @"octitle";
    
    button.backgroundColor = [UIColor redColor];
    [button sizeToFit];
    
    [self.view addSubview:button];
    
    
}

@end

File: Swift/Swift3基础/11-OC&Swift互调/SwiftCallOC/SwiftCallOC/ViewController.swift
//
//  ViewController.swift
//  SwiftCallOC
//
//  Created by 黄继平 on 2016/11/20.
//  Copyright © 2016年 PJStudio. All rights reserved.
//

import UIKit

/*
    1.在Swift项目中创建OC文件时,直接按照Xcode提示,生成一个bridge.h文件;
        1.1 或者自己手动创建一个bridge.h文件, 创建完成后,在项目->build setting中搜bridging->在bridging header对应的value中写入桥接文件的路径(项目名/桥接文件)
    2.在桥接文件中导入需要创建的OC头文件
    3.在Swift中就可以使用了!
 */
class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        
    }

    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        
        let button = CustomButton(type: .contactAdd)
        button.backgroundColor = UIColor.red
        
        view.addSubview(button)
        
    }

}

